server/src/task/test/task.e2e-spec.ts

```diff
@@ -0,0 +1,170 @@
+import { INestApplication, ValidationPipe } from '@nestjs/common';
+import * as request from 'supertest';
+import * as cookieParser from 'cookie-parser';
+import { TestingModule, Test } from '@nestjs/testing';
+import { AuthGuard } from '../../auth/guard';
+import { authSetTokens, buildDefaultModules } from '../../../test';
+import { TaskModule } from '../task.module';
+import {
+  validCreateTaskField,
+  invalidCreateTaskFields,
+  invalidUpdateTaskFields,
+  validUpdateTaskField,
+} from './data';
+
+describe('TaskController (e2e)', () => {
+  let app: INestApplication;
+
+  let token: string;
+  let csrfToken: string;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [...buildDefaultModules(), TaskModule],
+      providers: [AuthGuard],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe());
+    app.use(cookieParser());
+
+    await app.init();
+    app.setGlobalPrefix('api');
+
+    const tokens = await authSetTokens(app);
+    token = tokens.jwtToken;
+    csrfToken = tokens.csrfToken;
+  });
+
+  afterAll(() => {
+    app.close();
+  });
+
+  // create-task test
+
+  describe('POST /tasks/create', () => {
+    it('should return 401 if not authenticated', async () => {
+      await request(app.getHttpServer()).post(`/tasks/create`).expect(401);
+    });
+
+    // invalid test
+    invalidCreateTaskFields.forEach((invalidTaskField, index) => {
+      it(`should return 400 for invalid task fields - case ${index + 1}`, async () => {
+        await request(app.getHttpServer())
+          .post('/tasks/create')
+          .send(invalidTaskField)
+          .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+          .set('x-csrf-token', csrfToken)
+          .expect(400);
+      });
+    });
+
+    // valid test
+    it('should return 201 and the task data if authenticated', async () => {
+      await request(app.getHttpServer())
+        .post('/tasks/create')
+        .send(validCreateTaskField)
+        .set('Cookie', [`token=${token}`, `csrf-token=${csrfToken}`])
+        .set('x-csrf-token', csrfToken)
+        .expect(201);
```

Comment:
Could you also add the data check like I explained before? Please also add the other's success check in this file!

Reply:
Oh I forgot!
I added!
https://github.com/HAK2024/fairy-share/pull/69/commits/8caaf7d72179207883abc35ff54fec91426fc67c